### DISCUSSION TITLE
What's the memory space of host function, Is this a workable way to implement zero copy?

### DISCUSSION THREAD

**Discussion by ActivePeter** [WasmEdge/WasmEdge#3104] [Q&A]
I want to implement zero copy by offer host functions to wasm functions. The basic idea is as follows:
- offer vec_new and vec_push host function
- get vec and operate vec by vec_new and vec_push function
- return the final vec ptr

But this idea is based on the hypothesis that host function is running out of vm, then the allocation is out of vm, so the outer system will not need to copy vm memories to outer memory.

**Comment by hydai** [comment]
You can perform unrestricted actions within the host function, as it is not confined to the sandbox environment. This allows you to implement your own memory management system.

**Reply by ActivePeter** [reply]
Can the wasm lib access memory address in host

**Reply by hydai** [reply]
Wasm itself can only access its own memory instance. You have to handle all of the host memory inside the host functions.

**Comment by ActivePeter** [comment]
How to get vec result from host?
